survival_lib.py

```
@@ -176,63 +213,147 @@ def fit_km_logrank(df, duration_col, event_col, group_col):
             fig.add_trace(go.Scatter(
-                x=kmf.survival_function_.index,
-                y=kmf.survival_function_.iloc[:, 0],
-                mode='lines',
-                name=label,
-                line=dict(color=colors[i % len(colors)], width=2),
-                hovertemplate=f'{label}<br>Time: %{{x:.1f}}<br>Surv: %{{y:.3f}}<extra></extra>'
+                x=np.concatenate([ci_times, ci_times[::-1]]),
+                y=np.concatenate([ci_lower, ci_upper[::-1]]),
+                fill='toself',
+                fillcolor=rgba_color,
+                line=dict(color='rgba(255,255,255,0)'),
```

Ruff (0.14.10)

Unnecessary `dict()` call (rewrite as a literal)

Rewrite as a literal

(C408)